lightstep/legacy_query_equivalence.go

```diff
@@ -0,0 +1,151 @@
+package lightstep
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/lightstep/terraform-provider-lightstep/client"
+)
+
+// Checks if the prior dashboard attributes have legacy queries that
+// map to the UQL queries that the API call returns.
+//
+// If so, we can ignore the "false diff" a Terraform plan would otherwise show
+// by using the legacy resource data instread of the UQL resource data.
+func dashboardHasEquivalentLegacyQueries(
+	ctx context.Context,
+	c *client.Client,
+	projectName string,
+	priorAttrs *client.UnifiedDashboardAttributes,
+	updatedAttrs *client.UnifiedDashboardAttributes,
+) (bool, error) {
+	// This code is only applicalbe if there are legacy charts
+	for _, chart := range priorAttrs.Charts {
```

Comment:
do we care about catching cases where a dashboard has multiple charts where some are TQL and some are legacy? i think this will short circuit if any chart on the dashboard has a tql query